FinSearchUnified/Tests/Functional/Controllers/Frontend/FinSearchUnified_Tests_Controllers_Frontend_SearchTest.php

```diff
@@ -0,0 +1,344 @@
+            }
+        }
+        if ($didYouMeanQuery !== null) {
+            $query->addChild('didYouMeanQuery', $didYouMeanQuery);
+        }
+        if ($originalQuery !== null) {
+            $query->addChild('originalQuery', $originalQuery);
+        }
+
+        $results = $xmlResponse->addChild('results');
+        $results->addChild('count', 5);
+        $products = $xmlResponse->addChild('products');
+
+        for ($i = 1; $i <= 5; $i++) {
+            $product = $products->addChild('product');
+            $product->addAttribute('id', $i);
+        }
+
+        $httpResponse = new Zend_Http_Response(
+            200,
+            [],
+            '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>'
+        );
+        $urlBuilderMock = $this->getMockBuilder(UrlBuilder::class)
+            ->disableOriginalConstructor()
+            ->setMethods([
+                'buildCompleteFilterList',
+                'buildCategoryUrlAndGetResponse',
+                'setCustomerGroup',
+                'buildQueryUrlAndGetResponse'
+            ])
+            ->getMock();
+        if ($originalQuery !== null) {
+            $urlBuilderMock->expects($this->exactly(1))->method('buildCompleteFilterList')->willReturn($httpResponse);
+            $urlBuilderMock->expects($this->never())->method('buildCategoryUrlAndGetResponse');
+            $urlBuilderMock->expects($this->exactly(2))->method('setCustomerGroup');
+            $urlBuilderMock->expects($this->once())->method('buildQueryUrlAndGetResponse')->willReturn(
+                new Zend_Http_Response(200, [], $xmlResponse->asXML())
+            );
+        }
+
+        $facetGateway = new FindologicFacetGateway(
+            Shopware()->Container()->get('shopware_storefront.custom_facet_gateway'),
+            $urlBuilderMock
+        );
+
+        Shopware()->Container()->set('FinSearchUnified.findologic_facet_gateway', $facetGateway);
+
+        $productNumberSearch = new ProductNumberSearch(
+            Shopware()->Container()->get('shopware_search.product_number_search'),
+            $urlBuilderMock
+        );
+
+        Shopware()->Container()->set('fin_search_unified.product_number_search', $productNumberSearch);
+
+        $session = $this->getMockBuilder(Enlight_Components_Session_Namespace::class)
+            ->setMethods(['offsetGet', 'offsetExists'])
+            ->getMock();
+        $session->expects($this->atLeastOnce())->method('offsetExists')->willReturnMap([
+            'findologicDI' => true
+        ]);
+        $session->expects($this->atLeastOnce())->method('offsetGet')->willReturnMap([
+            ['isSearchPage', true],
+            ['isCategoryPage', false],
+            ['findologicDI', false]
+        ]);
+
+        Shopware()->Container()->set('session', $session);
+
+        $configArray = [
+            ['ActivateFindologic', true],
+            ['ActivateFindologicForCategoryPages', false],
+            ['findologicDI', false],
```

Comment:
`findologicDI`, `isSearchPage` and `isCategoryPage` are only available from the session.